lib/src/providers/VernaProvider/index.tsx

```diff
@@ -107,56 +94,66 @@ function useVerna() {
 }
 
 export interface VernaProviderProps extends Pick<FormProps<unknown>, 'transformErrors'> {
-  DropZone: React.FunctionComponent<DropZoneProps>;
+  DropZone: React.FunctionComponent<DropZoneOverloadProps>;
   FieldTemplate: React.FunctionComponent<FieldTemplateProps>;
   Section: React.FunctionComponent<ObjectFieldTemplateProps>;
   SubmitButton: React.ReactNode;
   configSchema?: VernaJSONSchemaType;
   defaultFormValues?: any;
   defaultLocale?: string;
-  defaultSchema: VernaJSONSchemaType;
   defaultSelector?: string;
-  defaultUiSchema: UiSchema;
   defaultWidgets: WidgetsType;
+  initialSchema: VernaSchemaType;
   intl?: ResolvedIntlConfig;
   isEditor: boolean;
   locale?: string;
-  translations: TranslationType;
+}
+
+function schemaReducer(state: VernaSchemaType, action: ReducerAction): VernaSchemaType {
+  const newSchema = _.cloneDeep(state);
+
+  // manage selector here with action.payload?.selector
+  if (action.error) {
+    console.error('An error happened while applying a modification on the schema');
+    console.error(action.payload);
+    return state;
+  } else {
+    return applyReducerAction(newSchema, action);
+  }
 }
 
 function VernaProvider({
   children,
   configSchema,
   defaultFormValues,
   defaultLocale,
-  defaultSchema,
   defaultSelector,
-  defaultUiSchema,
   defaultWidgets,
+  initialSchema,
   intl,
   isEditor,
   locale,
   SubmitButton,
-  DropZone,
+  DropZone: DropZoneRender,
   FieldTemplate,
   Section,
   transformErrors,
-  translations,
 }: PropsWithChildren<VernaProviderProps>) {
-  // Both fullSchema & fullUiSchema are not used by the lib itself but may be
-  // used for further implementation
-  const [fullSchema, setFullSchema] = useState<VernaJSONSchemaType>(defaultSchema);
-  const [fullUiSchema, setFullUiSchema] = useState<UiSchema>(
-    cleanUiSchema(defaultSchema, defaultUiSchema, Section),
+  const [schema, _schemaDispatch] = useReducer<Reducer<VernaSchemaType, ReducerAction>>(
+    schemaReducer,
+    initialSchema,
   );
+  const [selector, setSelector] = useState<string | undefined>(defaultSelector);
 
-  const [schema, _setSchema] = useState<VernaJSONSchemaType>(
-    getSelectedSchema(defaultSchema, defaultSelector),
-  );
-  const [uiSchema, _setUiSchema] = useState<UiSchema>(
-    getSelectedUiSchema(defaultUiSchema, defaultSelector),
-  );
-  const [schemaTranslations, setSchemaTranslations] = useState<TranslationType>(translations);
+  // Overload of the schema dispatch to add the selector
+  function schemaDispatch(action: ReducerAction) {
+    const dispatchAction = _.cloneDeep(action);
+    dispatchAction.payload.selector = selector;
+    _schemaDispatch(dispatchAction);
+  }
```

Comment:
We could use `meta` property to pass the selector.

```suggestion
function schemaDispatch(action: ReducerAction) {
    _schemaDispatch({...action, meta: { ...action.meta, selector }});
}
```